src/pyedb/dotnet/edb_core/components.py

```diff
@@ -2071,8 +2070,8 @@ def set_solder_ball(
             pin1 = list(cmp.pins.values())[0].pin
             pin_layers = pin1.GetPadstackDef().GetData().GetLayerNames()
             pad_params = self._padstack.get_pad_parameters(pin=pin1, layername=pin_layers[0], pad_type=0)
-            _sb_diam = min([self._get_edb_value(val).ToDouble() for val in pad_params[1]])
-            sball_diam = _sb_diam
+            _sb_diam = abs(min([self._get_edb_value(val).ToDouble() for val in pad_params[1]]))
+            sball_diam = 0.8 * _sb_diam
```

Comment:
Why do you multiply by 0.8 ?

Reply:
this is an arbitrary value based on experiment. When users export to 3D modeler having the same value cause trouble with meshing. Using a smaller value prevent this issue